### ISSUE TITLE
Configuration Migration Utility and Notes (Portal ALM)

### ISSUE THREAD

**Issue by lokiRT** [MicrosoftDocs/powerapps-docs#2251]
[Enter feedback here]
When filtering records with the Configuration Migration Utility, if the parent record is not part of an export, the import will skip annotation records with the error "PrimaryId Field annotationid and PrimaryName Field subject - FAILED with Error : The parent object type was present, but the ID was missing." although the data xml contains the objectid and type.  This looks to be due to the formation of the lookup for the api call, but might be looking for the corresponding record within the same data.xml to get the primary attribute value or id/type from there?  Our client is looking to filter the portal configuration changes in deploys to date ranges to avoid larger portal deployments, and we ran across this issue.  

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 099cc915-21d6-ca85-ecd6-c788c632b172
* Version Independent ID: 5d57e6cd-4f77-7775-83a7-621ae32134dc
* Content: [Download tools from NuGet (Microsoft Dataverse) - Power Apps](https://docs.microsoft.com/en-us/powerapps/developer/data-platform/download-tools-nuget)
* Content Source: [powerapps-docs/developer/data-platform/download-tools-nuget.md](https://github.com/MicrosoftDocs/powerapps-docs/blob/main/powerapps-docs/developer/data-platform/download-tools-nuget.md)
* Service: **powerapps**
* GitHub Login: @shmcarth
* Microsoft Alias: **jdaly**

**Comment by lokiRT** [comment]
adx_webformstep seem to have this same issue with next step and previous step, so this looks to be any lookup to a record that isn't already in the exported data potentially.  My expectation is that means there is a validation step checking the import data.xml file and removing any lookups where the reference is not found in the file.  The workaround options we have found while still keeping this minimal (still using date filters for new/modified records) expands the list to include related records via linked entities, and before there was filtering in Configuration Migration we wrote custom PowerShell to do this.  

I have attached a sample filter for the initial find, which includes any webfiles created or updated within a date range as well as any webfiles where a related note was created or updated in the date range.  I have had multiple customers looking to have delta deploys using date ranges like this, specifically with portals.  Even a switch for the import disabling the import lookup validations, similar to the "do not update existing records for all entities" and "disable plug-ins on all entities for import" would allow deltas to bypass this issue.

[webfile filter with related notes.txt](https://github.com/MicrosoftDocs/powerapps-docs/files/6282498/webfile.filter.with.related.notes.txt)
